src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.ts

```diff
@@ -167,7 +178,10 @@ export default class AmazonPayPaymentStrategy implements PaymentStrategy {
                 },
                 onPaymentSelect: orderReference => {
                     this._synchronizeBillingAddress()
-                        .then(() => onPaymentSelect(orderReference))
+                        .then(() => {
+                            this._isPaymentMethodSelected = true;
```

Comment:
I believe `onPaymentSelect` gets called each time the shopper selects a new credit card option. Does that mean once it's set to `true`, there's a chance that something could fail later and the flag should be unset?

Reply:
Yes you are right `onPaymentSelect` is called each time the shopper selects a new credit card option. Actually this flag was added in order to be able to identify when a credit card is not selected. This scenario occurs when the wallet widget is loaded after a purchase failed with an `InvalidPaymentMethod` code